File: KantechGatewayApp/Infrastructure/SimpleScheduler.vb
Namespace KantechGatewayApp.Infrastructure
    Public Class SimpleScheduler
        Private ReadOnly _mgr As JobManager
        Private ReadOnly _timer As Timers.Timer

        Public Sub New(mgr As JobManager, intervalSec As Integer)
            _mgr = mgr
            _timer = New Timers.Timer(intervalSec * 1000)
            AddHandler _timer.Elapsed, AddressOf OnTick
            _timer.AutoReset = True
        End Sub

        Public Sub Start()
            _timer.Start()
        End Sub
        Public Sub [Stop]()
            _timer.Stop()
        End Sub

        Private Sub OnTick(sender As Object, e As Timers.ElapsedEventArgs)
            _mgr.MaybeRunScheduled()
        End Sub
    End Class
End Namespace

File: KantechGatewayApp/Infrastructure/SimpleScheduler.cs

namespace KantechGatewayApp.KantechGatewayApp.Infrastructure
{
    public class SimpleScheduler
    {
        private readonly JobManager _mgr;
        private readonly System.Timers.Timer _timer;

        public SimpleScheduler(JobManager mgr, int intervalSec)
        {
            _mgr = mgr;
            _timer = new System.Timers.Timer(intervalSec * 1000);
            _timer.Elapsed += OnTick;
            _timer.AutoReset = true;
        }

        public void Start()
        {
            _timer.Start();
        }
        public void Stop()
        {
            _timer.Stop();
        }

        private void OnTick(object sender, System.Timers.ElapsedEventArgs e)
        {
            _mgr.MaybeRunScheduled();
        }
    }
}